src/server/managers/AppExternalComponentManager.ts

```diff
@@ -1,79 +1,129 @@
 import { IExternalComponent } from '../../definition/externalComponent';
+import { ExternalComponentAlreadyTouchedError, ExternalComponentNotMatchWithAppError } from '../errors';
 
 /**
- * The external component manager for the Apps.
+ * The external component manager for the apps.
  *
- * An App will register an external component during their `initialize` method.
- * Then once an App's `onEnable` is called and it returns true,
- * only then will that App's external component be enabled.
+ * An app will register external components during its `initialize` method.
+ * Then once an app's `onEnable` method is called and it returns true,
+ * only then will that app's external components be enabled.
  */
 export class AppExternalComponentManager {
     /**
-     * The map that maintain the list of all registered components.
+     * The map that maintains all registered components.
+     * The key of the top map is app id and the key of inner map is the
```

Comment:
With this addition, you may want to modify the documentation on the `IExternalComponent` to state that the `name` must be unique.